x/data/features/msg_anchor.feature

```diff
@@ -0,0 +1,54 @@
+Feature: MsgAnchor
+
+  Rule: only a valid sender is accepted
+
+    Scenario: an error is returned if sender is empty
```

Comment:
As is this is more informative than the rule above

Reply:
If the rule is "an error is returned if sender is empty" then the scenarios would be "sender is empty", "sender is not empty", and "sender is not empty" could still return an error if not a valid address.

Reply:
This would also cause a lot of duplication if we were testing the valid case for each rule. "sender is not empty" (or sender is a valid address) would then be duplicated with a different scenario name for each rule.

Reply:
So maybe we shouldn't use Rule?

Reply:
Hmmm... I'm not sure at this point. I think Rule could be useful in some cases. It might make sense in the context of the server methods but not necessarily for message and content hash validation. For example, when we are testing whether or not the data has been anchored when we are actually testing the `Attest` or `RegisterResolver` methods.

Reply:
I think Rule could be useful in some cases. It might make sense in the context of the server methods but not necessarily for message and content hash validation. For example, when we are testing whether or not the data has been anchored when we are actually testing the `Attest` or `RegisterResolver` methods ([see here](https://github.com/regen-network/regen-ledger/pull/994/files#diff-ee6cd2c2c155f81315035619ed96293656b72688fc7597e3399ca43a8cfd8df5R15-R41)).